Documentation/release-notes/4054.md

```diff
@@ -0,0 +1,24 @@
+## Support for Binding Kotlin Libraries
+
+This release adds better support for creating bindings for Kotlin libraries, including:
+
+* Hiding Kotlin `internal` types and members
+* Using Kotlin provided metadata to properly name method parameters
+* Hiding Kotlin generated methods like `foo-impl` that aren't intended for public use
+* Restoring Kotlin-mangled method names to their original name (ie: `foo-V5j3Lk8` -> `foo`)
```

Comment:
Just for me to get ahead of the game with putting the notes in the final docs style (I learned today that "[i.e.](https://docs.microsoft.com/style-guide/a-z-word-list-term-collections/i/ie)" and "[e.g.](https://docs.microsoft.com/style-guide/a-z-word-list-term-collections/e/eg)" are apparently discouraged):

```suggestion
* Restoring Kotlin-mangled method names to their original names, such as changing `foo-V5j3Lk8` back to `foo`
```